yt/fields/field_info_container.py

```diff
@@ -402,6 +404,12 @@ def add_output_field(self, name, sampling_type, **kwargs):
         kwargs.setdefault("ds", self.ds)
         self[name] = DerivedField(name, sampling_type, NullFunc, **kwargs)
 
+    def __setitem__(self, key, value):
+        ftype, fname = key
+        if any(fname == _fname for _ftype, _fname in self):
```

Comment:
It took me multiple passes to understand this bit. It would help to add a comment here
```suggestion
        # Mark fields with same `fname`s (but difference `ftype`s) as ambiguous
        if any(fname == _fname for _ftype, _fname in self):
```